### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by rinx**
Signed-off-by: Rintaro Okamura <rintaro.okamura@gmail.com>

This PR includes
- Add E2E workflow
- Add E2E test code
- Update dependencies
- Regenerate gRPC client code
- Revise release workflow

### CODE REVIEW

**Review by kpango — APPROVED** [review]
LGTM

**Review by kpango — APPROVED** [review]

**Review by kevindiu — APPROVED** [review]
LGTM

File: .github/actions/install-grpc-dependencies/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        cd $HOME/grpc-java/compiler/src/java_plugin/cpp
+        g++ -I. -I$HOME/protobuf/src \
+        *.cpp \
+        -L$HOME/protobuf/src/.libs \
+        -lprotoc -lprotobuf -lpthread --std=c++0x -s \
+        -o protoc-gen-grpc-java
+        cp protoc-gen-grpc-java $HOME/grpc-java
+      env:
+        PROTOBUF_VERSION: v3.17.3
+        GRPC_JAVA_VERSION: v1.39.0
```
Comment (kevindiu) [inline .github/actions/install-grpc-dependencies/action.yml:36]: BTW the latest version is 1.40 now.

Comment (rinx) [reply .github/actions/install-grpc-dependencies/action.yml:36]:
```suggestion
        GRPC_JAVA_VERSION: v1.40.0
```

**Comment by rinx** [comment]
Thanks for reviewing.

**Pull request merged**